### PR TITLE
Add support for Datadog logging

### PR DESCRIPTION
**Pull request by ezkl**
Adds support for configuring [Datadog log streaming](https://docs.fastly.com/en/guides/log-streaming-datadog) ([API docs](https://developer.fastly.com/reference/api/logging/datadog/)) to provider.

cc: @phamann @mccurdyc

### CODE REVIEW

File: fastly/block_fastly_service_v1_logging_datadog.go [hunk 1]
```diff
@@ -0,0 +1,210 @@
...
+				Optional:     true,
+				Default:      2,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either `1` or `2`. (default: `2`)",
+				ValidateFunc: validateLoggingFormatVersion(),
+			},
+
+			"placement": {
+				Type:         schema.TypeString,
+				Optional:     true,
+				Description:  "Where in the generated VCL the logging call should be placed",
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_datadog.go]: Include "Can be none or waf_debug"?

File: fastly/block_fastly_service_v1_logging_datadog.go [hunk 2]
```diff
@@ -0,0 +1,210 @@
...
+				Type:        schema.TypeString,
+				Optional:    true,
+				Description: "Apache-style string or VCL variables to use for log formatting.",
+			},
+
+			"format_version": {
+				Type:         schema.TypeInt,
+				Optional:     true,
+				Default:      2,
+				Description:  "The version of the custom logging format used for the configured endpoint. Can be either `1` or `2`. (default: `2`)",
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_datadog.go]: Include "The logging call gets placed by default in vcl_log if format_version is set to 2 and in vcl_deliver if format_version is set to 1."?

File: fastly/block_fastly_service_v1_logging_datadog.go [hunk 3]
```diff
@@ -0,0 +1,210 @@
...
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func deleteDatadog(conn *fst.Client, i *fst.DeleteDatadogInput) error {
+	err := conn.DeleteDatadog(i)
+
+	if errRes, ok := err.(*fst.HTTPError); ok {
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_datadog.go]: Same as other PRs, but still noting, wouldn't this result in 404s returning nil? Is this expected?

File: fastly/block_fastly_service_v1_logging_datadog_test.go [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+    name   = "datadog-endpoint"
+    token  = "t0k3n"
+    region = "EU"
+    format = "%%h %%l %%u %%t \"%%r\" %%>s %%b %%T"
+  }
+
+  logging_datadog {
+    name  = "another-datadog-endpoint"
+    token = "another-token"
+  }
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_datadog_test.go:304]: Why doesn't this second logging endpoint have the value included that was set here

https://github.com/terraform-providers/terraform-provider-fastly/pull/242/files#diff-952caa4401e24987ac6e257a290b1f32R161

File: website/docs/r/service_v1.html.markdown [hunk 5]
```diff
@@ -517,6 +519,16 @@ The `httpslogging` block supports:
...
 
+The `logging_datadog` block supports:
+
+* `name` - (Required) The unique name of the Datadog logging endpoint
+* `token` - (Required) The API key from your Datadog account.
+* `region` - (Optional) The region that log data will be sent to. One of `US` or `EU`. Defaults to `US` if undefined.
+* `format` - (Optional) Apache-style string or VCL variables to use for log formatting.
+* `format_version` - (Optional) The version of the custom logging format used for the configured endpoint. Can be either `1` or `2`. (default: `2`)
+* `placement` - (Optional) Where in the generated VCL the logging call should be placed.
+* `response_condition` - (Optional) The name of the condition to apply.
```
Comment (mccurdyc) [inline website/docs/r/service_v1.html.markdown:589]: Same notes on docs as above.

**Review by mccurdyc — COMMENTED** [review]
Notes on docs and one question about a test case not including a value that I would have expected.

**Review by phamann — APPROVED** [review]
:+1: LGTM - you'll need to rebase/fix merge conflicts

**Comment by mccurdyc** [comment]
Force-pushed rebasing master after merging - https://github.com/terraform-providers/terraform-provider-fastly/pull/262

File: fastly/block_fastly_service_v1_logging_datadog_test.go [hunk 6]
```diff
@@ -0,0 +1,309 @@
...
+
+  force_destroy = true
+}
+`, name, domain)
+}
+
+var (
+	// https://www.terraform.io/docs/configuration/expressions.html#string-literals
+	escapePercent          = strings.ReplaceAll(datadogDefaultFormat, "%", "%%")
+	escapeTemplateSequence = strings.ReplaceAll(escapePercent, "%%{", "%%%%{")
```
Comment (mccurdyc) [inline fastly/block_fastly_service_v1_logging_datadog_test.go:273]: This took me a while to figure out :(

**Comment by mccurdyc** [comment]
Force-pushed squashing  test fix commits into a single commit.

**Pull request merged**